presto-hive/src/main/java/io/prestosql/plugin/hive/util/Statistics.java

```diff
@@ -117,63 +123,74 @@ public static HiveColumnStatistics merge(HiveColumnStatistics first, HiveColumnS
         if (first.isPresent() && second.isPresent()) {
             return Optional.of(new IntegerStatistics(
                     reduce(first.get().getMin(), second.get().getMin(), MIN, true),
                     reduce(first.get().getMax(), second.get().getMax(), MAX, true)));
         }
-        return Optional.empty();
+
+        return firstPresentOrEmpty(first, second);
     }
 
     private static Optional<DoubleStatistics> mergeDoubleStatistics(Optional<DoubleStatistics> first, Optional<DoubleStatistics> second)
     {
-        // normally, either both or none is present
         if (first.isPresent() && second.isPresent()) {
             return Optional.of(new DoubleStatistics(
                     reduce(first.get().getMin(), second.get().getMin(), MIN, true),
                     reduce(first.get().getMax(), second.get().getMax(), MAX, true)));
         }
-        return Optional.empty();
+
+        return firstPresentOrEmpty(first, second);
     }
 
     private static Optional<DecimalStatistics> mergeDecimalStatistics(Optional<DecimalStatistics> first, Optional<DecimalStatistics> second)
     {
-        // normally, either both or none is present
         if (first.isPresent() && second.isPresent()) {
             return Optional.of(new DecimalStatistics(
                     reduce(first.get().getMin(), second.get().getMin(), MIN, true),
                     reduce(first.get().getMax(), second.get().getMax(), MAX, true)));
         }
-        return Optional.empty();
+
+        return firstPresentOrEmpty(first, second);
     }
 
     private static Optional<DateStatistics> mergeDateStatistics(Optional<DateStatistics> first, Optional<DateStatistics> second)
     {
-        // normally, either both or none is present
         if (first.isPresent() && second.isPresent()) {
             return Optional.of(new DateStatistics(
                     reduce(first.get().getMin(), second.get().getMin(), MIN, true),
                     reduce(first.get().getMax(), second.get().getMax(), MAX, true)));
         }
-        return Optional.empty();
+
+        return firstPresentOrEmpty(first, second);
     }
 
     private static Optional<BooleanStatistics> mergeBooleanStatistics(Optional<BooleanStatistics> first, Optional<BooleanStatistics> second)
     {
-        // normally, either both or none is present
         if (first.isPresent() && second.isPresent()) {
             return Optional.of(new BooleanStatistics(
                     reduce(first.get().getTrueCount(), second.get().getTrueCount(), ADD, false),
                     reduce(first.get().getFalseCount(), second.get().getFalseCount(), ADD, false)));
         }
-        return Optional.empty();
+
+        return firstPresentOrEmpty(first, second);
+    }
+
+    private static <T> Optional<T> firstPresentOrEmpty(Optional<T> left, Optional<T> right)
+    {
+        verify(!(left.isPresent() && right.isPresent()), "either left optional or right should not be present");
```

Comment:
Why?

method name suggests it should return the first present, even if both are present